storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10414,15 +10467,33 @@ int ha_rocksdb::inplace_populate_sk(
       }
 
       /* Create new secondary index entry */
+
+      String value_buffer;
```

Comment:
hmm, not sure how I feel about this.  I think it makes the code change simpler, but now we are making a new string for every secondary index entry that's getting added and appending (allocing memory) to it.  

Have we considered, passing the info we need into`pack_record` and attempting to build the record in-place via the `Rdb_string_writer`?